common/components/controllers/CorporateHackathonController.jsx

```diff
@@ -0,0 +1,219 @@
+import url from "../utils/url.js";
+import Button from "react-bootstrap/Button";
+import Section from "../enums/Section";
+import Ch1Icon from "../svg/corporatehackathon/one.svg";
+import Ch2Icon from "../svg/corporatehackathon/two.svg";
+import Ch3Icon from "../svg/corporatehackathon/three.svg";
+import cdn from "../utils/cdn.js";
+
+type State = {|
+  showContactModal: boolean
+|};
+
+class CorporateHackathonController extends React.PureComponent<{||}, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showContactModal: false
+    };
+  }
+
+  componentDidMount() {
+    prerender.ready();
+  }
+
+  _renderHeader(): React$Node {
+    const title: string = "DemocracyLab | Corporate Hackathons";
+    const description: string = "Host a hackathon with DemocracyLab!"
+
+    return (
+      <Headers
+        title={title}
+        description={description}
+      />
+    );
+  }
+
+  onSubmit(): void {
+    this.setState({showContactModal: false});
+  }
+
+   render(): $React$Node {
+     return (
+       <React.Fragment>
+         <ContactUsModal showModal={this.state.showContactModal} onSubmit={this.onSubmit.bind(this)} />
+         {this._renderHeader()}
+        <div className="corporate-hackathon-root">
+          {this._hostWithUs()}
+          <div className="container">
+            <div className="row">
+              {this._howItWorks()}
+              {this._whyEngage()}
+              {this._strengthenTeam()}
+              {this._hackathonSponsorships()}
+            </div>
+          </div>
+        </div>
+       </React.Fragment>
+     )
+   }
+
+  _hostWithUs(): $React$Node {
+    return (
+      <div className="corporate-hackathon-hostwithus hero-wide ch-divider">
+        <div className="hero-wide-content">
+          <div className="hero-wide-flex-content">
+            <div className="corporate-hackathon-hostwithus-left">
+              <h1>Host a hackathon with us!</h1>
+              <p className="corporate-hackathon-hostwithus-left-text">Create a unique employee engagement experience and deliver instant-impact tech projects.</p>
+              <Button className="corporate-hackathon-hostwithus-left-button" variant="primary" onClick={() => this.setState({showContactModal: true})}>Get Started</Button>
+              {this._publicHackathonResults("d-lg-none d-xl-block")}
+            </div>
+            <div className="corporate-hackathon-hostwithus-right">
+              <div className="corporate-hackathon-hostwithus-header">
+                <picture>
+                  <source media="(max-width: 991px)" srcSet={cdn.image("mobile-header-withbg.png")} />
+                  <source media="(min-width: 992px)" srcSet={cdn.image("header-withbg-img.png")} />
+                  <img src={cdn.image("header-withbg-img.png")} alt="Header Image" />
+                </picture>
+              </div>
+            </div>
+          </div>
+          {this._publicHackathonResults("d-none d-lg-block d-xl-none")}
```

Comment:
Why are we rendering this element twice?

Reply:
A very inelegant way to handle the public hackathon results list either being in the left column of the section (thus constrained by the width of that left column) or at certain breakpoints being broken out 'below' and being allowed to take 100%.

the `d-*` classes are Bootstrap helper classes denoting at which breakpoints each render is allowed to be visible (by toggling display: none and block, basically)